python/arcticdb/adapters/storage_fixture_api.py

```diff
@@ -0,0 +1,155 @@
+"""
+Copyright 2023 Man Group Operations Limited
+
+Use of this software is governed by the Business Source License 1.1 included in the file LICENSE.txt.
+
+As of the Change Date specified in that file, in accordance with the Business Source License, use of this software will be governed by the Apache License, version 2.0.
+"""
+
+import multiprocessing
+import subprocess
+import os
+import platform
+import signal
+import functools
+import socketserver
+from typing import Dict, Union, Pattern
+from contextlib import AbstractContextManager, contextmanager
+from abc import abstractmethod
+from enum import Enum
+
+from arcticdb.config import Defaults
+from arcticdb.version_store import NativeVersionStore
+from arcticdb.version_store.helper import ArcticMemoryConfig
+from arcticdb.util.test import apply_lib_cfg
+from arcticc.pb2.storage_pb2 import EnvironmentConfigsMap
+
+_WINDOWS = platform.system() == "Windows"
+
+
+def get_ephemeral_port():  # https://stackoverflow.com/a/61685162/
+    with socketserver.TCPServer(("localhost", 0), None) as s:
+        return s.server_address[1]
+
+
+def start_process_that_can_be_gracefully_killed(cmd, **kwargs):
```

Comment:
Kind of long-winded. Maybe start_child_process and kill_child_process?

Reply:
The name hints it should be used with the `gracefully_kill_process` function below and `gracefully` indicates the child process is given a chance to clean up instead of being directly killed (matters to Windows, which has no SIGTERM).

After this is imported to the calling file, an unqualified name `start_child_process` may cause confusion and get accidentally replaced with plain `run` or `Popen`, which if used with `gracefully_kill_process` on Windows will kill all processes running from the same terminal including the test runner.